release_creation/github_client/releases.py

```diff
@@ -0,0 +1,87 @@
+import os
+import requests
+from typing import Dict, List, Optional, Set, Tuple
+from semantic_version import Version, match
+from github import Github
+from github.GitRelease import GitRelease
+from github.GitReleaseAsset import GitReleaseAsset
+
+_GH_SNAPSHOT_REPO = "dbt-labs/dbt-core-snapshots"
+_GH_ACCESS_TOKEN = os.environ["GH_ACCESS_TOKEN"]
+_SNAP_REQ_NAME = "snapshot_requirements"
+
+
+def get_github_client() -> Github:
+    return Github(_GH_ACCESS_TOKEN)
+
+
+def get_latest_release(input_version:str) -> Tuple[Version, Optional[GitRelease]]:
+    gh = get_github_client()
+    target_version = Version(input_version)
+    target_match = f"~{target_version.major}.{target_version.minor}"
+    latest = Version(major=target_version.major, minor=target_version.minor, patch=0)
+    repo = gh.get_repo(_GH_SNAPSHOT_REPO)
+    releases = repo.get_releases()
+    latest_release = None
+    for r in releases:
+        release_version = Version.coerce(r.tag_name)
+        if match(target_match, r.tag_name) and release_version > latest:
+            latest = release_version
+            latest_release = r
+    return latest, latest_release
```

Comment:
My understanding is that [get_latest_release](https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository.get_latest_release) doesn't serve our purposes here because we're looking for the latest patch release corresponding to the major.minor version specified by `input_version`, correct?

Reply:
the `match` function returns true on corresponding major/minor versions then we check if it's the latest within the major/minor by `release_version > latest`